presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java

```diff
@@ -62,7 +63,7 @@ public Optional<RecordCursor> createRecordCursor(
         // make sure the FileSystem is created with the proper Configuration object
         Path path = new Path(fileSplit.getPath());
         try {
-            if (!fileSplit.getCustomSplitInfo().isEmpty() && isHudiRealtimeSplit(fileSplit.getCustomSplitInfo())) {
+            if (!fileSplit.getCustomSplitInfo().isEmpty() && (isHudiRealtimeSplit(fileSplit.getCustomSplitInfo()) || isBootstrapBaseFileSplit(fileSplit.getCustomSplitInfo()))) {
```

Comment:
Ideally you need another check to see if it is a `HoodieRealtimeBootstrapBaseFileSplit`.

I would rather remove the existing `isHudiRealtimeSplit` check as well since if fileSplit.getCustomSplitInfo() is not empty, it is one of HoodieRealtimeFileSplit/BootstrapBaseFileSplit/HoodieRealtimeBootstrapBaseFileSplit. This concept of CustomSplitConverter was introduced for hudi tables only. Hence checking the split type is redundant.

Reply:
Makes sense. I have removed the condition to check the split type.